### PR TITLE
[Modules] Removed the paramerers of those properties that allow a single value only & updated docs

### PR DESCRIPTION
**Pull request by AlexanderSehr**
# Description

- Removed the paramerers of those properties that allow a single value only
- Updated docs
- Removed 'minPrem' test from storage account to unblock tests (and because the feature isn't available in our test location West Europe)

| Pipeline |
| - |
| [KeyVault - Vaults](https://github.com/Azure/ResourceModules/actions/workflows/ms.keyvault.vaults.yml) |
| [Resources - Tags](https://github.com/Azure/ResourceModules/actions/workflows/ms.resources.tags.yml) |
| [ServiceBus - Namespaces](https://github.com/Azure/ResourceModules/actions/workflows/ms.servicebus.namespaces.yml) |
| [Sql - ManagedInstances](https://github.com/Azure/ResourceModules/actions/workflows/ms.sql.managedinstances.yml) |
| [Storage - StorageAccounts](https://github.com/Azure/ResourceModules/actions/workflows/ms.storage.storageaccounts.yml) |

# Type of Change

Please delete options that are not relevant.

- [ ] Bugfix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [x] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Update to documentation

### CODE REVIEW

File: modules/Microsoft.Storage/storageAccounts/blobServices/deploy.bicep
Comment (eriqua) [inline modules/Microsoft.Storage/storageAccounts/blobServices/deploy.bicep:57]: I'd keep the optional diagnostic settings feature as aligned as possible across all modules. In this and other cases below I'd suggest using a variable `name` since the hardcoded value is used in more than one place

Comment (AlexanderSehr) [reply modules/Microsoft.Storage/storageAccounts/blobServices/deploy.bicep:57]: hm ok

Comment (AlexanderSehr) [reply modules/Microsoft.Storage/storageAccounts/blobServices/deploy.bicep:57]: Just tried and actually - no. We cannot reference variables in parameters.

Comment (eriqua) [reply modules/Microsoft.Storage/storageAccounts/blobServices/deploy.bicep:57]: Just opened a draft PR #2814 with what I meant

Comment (AlexanderSehr) [reply modules/Microsoft.Storage/storageAccounts/blobServices/deploy.bicep:57]: Looks good. The only 'caviat' is, that now the user is actually not aware of this default value as it is not published to the readme.

Comment (eriqua) [reply modules/Microsoft.Storage/storageAccounts/blobServices/deploy.bicep:57]: True, in that case we should probably add something in the description, e.g., "defaults to ... if empty"? In case this alternative makes sense, I guess it should be implemented in a consistent way also for modules not having an hardcoded name? Maybe worth a quick internal discussion.

Comment (AlexanderSehr) [reply modules/Microsoft.Storage/storageAccounts/blobServices/deploy.bicep:57]: Following a discussion we'll go with the suggestion by @eriqua to keep the diagnostic settings aligned & reduce the amount of duplicated code/values

Comment (AlexanderSehr) [reply modules/Microsoft.Storage/storageAccounts/blobServices/deploy.bicep:57]: We can resolve this comment once the PR was merged into this

File: modules/Microsoft.Storage/storageAccounts/queueServices/readme.md
Comment (eriqua) [inline modules/Microsoft.Storage/storageAccounts/queueServices/readme.md:38]: I guess this should be updated as well?

Comment (AlexanderSehr) [reply modules/Microsoft.Storage/storageAccounts/queueServices/readme.md]: good point. Done

Comment (AlexanderSehr) [reply modules/Microsoft.Storage/storageAccounts/queueServices/readme.md]: This will become real interesting for the Rest2Carml automation :D But I think by now with some of the constructs we have (like the necessary use of union) we cannot use it to update much anymore anyways :o

File: docs/wiki/The library - Module design.md
Comment (eriqua) [inline docs/wiki/The library - Module design.md:468]: I'm not sure I fully grasp the expression here. Examples make it clear, but what does the * stand for? Shouldn't the schema be `<mainResourceType>_<referencedResourceType>`?

File: docs/wiki/The library - Module design.md
Comment (eriqua) [inline docs/wiki/The library - Module design.md:395]: Far from easy, but should we update the paragraph to include the updated approach i.e., using the variable name and make the diagnosticName empty?

Comment (eriqua) [reply docs/wiki/The library - Module design.md:395]: I'm also thinking that using a variable for the hardcoded value even when diagnostic is not supported would be easier to describe and provide guidelines for. I sense you won't agree, passing you the ball😜

File: docs/wiki/The library - Module design.md
Comment (AlexanderSehr) [reply docs/wiki/The library - Module design.md:468]: This was some damn extension. And I still did not find out which. This must be undone

File: docs/wiki/The library - Module design.md
Comment (AlexanderSehr) [reply docs/wiki/The library - Module design.md:395]: I definitely wouldn't agree doing it 😉 but yeah, describing it would be easier 😄
All add a "special case" entry here. I hope thought that we won't end up with too many in the long term. Already many contributions seem to ignore the existing ones

Comment (AlexanderSehr) [reply docs/wiki/The library - Module design.md:395]: Done

Comment (AlexanderSehr) [reply docs/wiki/The library - Module design.md:395]: For 'all' it will anyways become quite interesting as I believe for some resources we already apply some logic to the name which we now need to intertwine with this.

Comment (AlexanderSehr) [reply docs/wiki/The library - Module design.md:395]: So I'm not sure how to update the description for that. Or do you already have something in mind?
I can also update the other diagnostic settings in this PR if you want :)

Comment (eriqua) [reply docs/wiki/The library - Module design.md:395]: Let's keep them separated as this PR is done already (will just have a final review). How to update the wiki will become more clear once we update others. This is correct for the current implementation so I think is good

**Review by eriqua — APPROVED** [review]
🆗

**Pull request merged**
